docs/userguide/campaigns.md

```diff
@@ -1,25 +1,146 @@
+When constructing a campaign, it is necessary to provide two objects:
+A search space (see [class](baybe.searchspace.core.SearchSpace) resp. [user guide](./searchspace))
+and an optimization objective (see [class](baybe.objective.Objective) resp. [user guide](./objective)).
+Optionally, it is possible to provide a different strategy
+(see [class](baybe.strategies.base.Strategy) resp. [user guide](./strategy)) and specify
+other aspects of the campaign (see [here](#AM) for details on
+`numerical_measurements_must_be_within_tolerance`).
 
-The following additional aspects are available to further specify a campaign:
-* **A strategy:**: The strategy that is used during the campaign. As a default, a [`TwoPhaseStrategy`](baybe.strategies.composite.TwoPhaseStrategy) is employed. For more details on strategies, see [here](./strategy).
-* **Conducted experiments**: A pandas ``DataFrame`` containing the experimental representation of previously conducted experiments. Preferably, the entries of this ``DataFrame`` were created by a previous campaign using [the campaigns recommend function](baybe.campaign.Campaign.recommend). If no such ``DataFrame`` is provided, it is assumed that no experiments were conducted previously.
-* **Numerical tolerance**: This is a flag for forcing numerical measurements to be within a pre-defined tolerance. Note that the setting of the tolerances is controlled as a part of the respective parameter.
-* **Previously done batches and fits** In case that a campaign builds upon previously condiucted experiments, it is possible to provide the number of previously done batches and fits.
 
-## How to use campaigns
+~~~python
+from baybe import Campaign
 
-We have a detailed example explaining how to use the ``Campaign`` object in our examples, so please see [here](./../../examples/Basics/campaign) for all necessary information.
\ No newline at end of file
+campaign = Campaign(
+    searchspace=searchspace,  # Required
+    objective=objective,  # Required
+    strategy=strategy,  # Optional
+    numerical_measurements_must_be_within_tolerance=boolean,  # Optional
+)
+~~~
+
+```{attention}
+Note that we currently also expose other fields via the constructor. This is
+only temporary, and the corresponding fields should be ignored.
+```
+
+### Creating a campaign via a JSON config
+
+It is also possible to specify a `Campaign` via a configuration string and using the
+function [`Campaign.from_config`](baybe.campaign.Campaign.from_config).
+The specification of config files as well as what they contain are automatically
+derived from the class structure.
+Furthermore, a config file can be validated using 
+[Campaign.validate_config](baybe.campaign.Campaign.validate_config).
+For more details and a full exemplary config, we refer to the corresponding
+[example](./../../examples/Serialization/create_from_config).
+
+## Getting recommendations
+
+### Basics
+
+```{attention}
```

Comment:
I am not sure if I like the fact that this block is there. I know, there has been confusion about the constructor signature and the attributes being mutable and public, but I think this is really something that we should simply fix soon and not mention here. Effectively, this is like offering a bad interface and then informing people not to abuse it. Instead of doing the latter, we should just fix the interface and not bother.  Similarly, we wouldn't tell for our other modules that the users should not fiddle with the internals!?

If I was a new user and read "the only safe way", then I would seriously question the design of the code :D

Reply:
So what would you prefer? Would a more defensive version work for you until this is fixed? Imagining something like (bold type just here to make spotting changes easier)
"Obtaining recommendations resp. adding measurements using the `recommend` resp.
`add_measurements` functions is the **currently intended** way to inform a `Campaign` object about
new measurements. These functions update the necessary metadata that is crucial for the
proper execution of a campaign. **We recommend** to rely on these functions to maintain
the integrity and reliability of the campaign's execution."

Reply:
we can always remove the box once the stuff is actually fixed, which I assume youre not suggesting to do in this PR?

Reply:
So what about including a more defensive formulation, explicitly recommending how it is intended, stating that un-intended use might break the object's inner logic and that we plan to improve this aspect to prevent users from misusing this?

Reply:
For my taste, I would simply remove the box altogether, but if you both prefer having it until the issues are really fixed, then let's keep the defensive formulation. Here a small fix then, because the first sentence is misleading (`recommend`  does not inform the campaign about new measurements) and these are methods, not functions:

Requesting recommendations via `recommend` and adding measurements via `add_measurements` is the only intended way to interact with a `Campaign` object. These methods update the necessary metadata that is crucial for the proper execution of a campaign. We recommend to rely on these methods to maintain
the integrity and reliability of the object.

Reply:
Thanks, will include your formulation!